index.js

```diff
@@ -39,19 +39,21 @@ async function run(context, plugins) {
     });
   }
 
-  if (isCi && isPr && !options.noCi) {
+  if (isCi && isPr && !options.noCi && !options.skipPrCheck) {
     logger.log("This run was triggered by a pull request and therefore a new version won't be published.");
     return false;
   }
 
-  if (ciBranch !== options.branch) {
+  if (![ciBranch, '*'].includes(options.branch)) {
     logger.log(
       `This test run was triggered on the branch ${ciBranch}, while semantic-release is configured to only publish from ${
         options.branch
       }, therefore a new version won’t be published.`
     );
     return false;
   }
+  options.branch = isPr ? prBranch : ciBranch;
```

Comment:
This should probably be moved to [L46](https://github.com/semantic-release/semantic-release/pull/1076/files#diff-168726dbe96b3ce427e7fedce31bb0bcR46)